libdevcore/CommonData.h

```diff
@@ -159,6 +161,17 @@ inline std::string toCompactBigEndianString(_T _val)
 	return ret;
 }
 
+/// Convenience function for conversion of a u256 to hex
+inline std::string toHex(u256 val, HexPrefix prefix = HexPrefix::DontAdd)
+{
+	std::string str = toHex(toBigEndian(val));
+	return (prefix == HexPrefix::Add) ? "0x" + str : str;
+}
+inline std::string toCompactHex(u256 val, HexPrefix prefix = HexPrefix::DontAdd)
+{
+	std::string str = toHex(toCompactBigEndian(val, 1)); //1 means val=0 would be '0x00' instead of '0x'
```

Comment:
This now calls the toHex function in line 164 instead if line 53 (what you want). And because of the toBigEndian there, you got 0x000..0001 instead of 0x01.